commit dd3668ec31a66381f9c7f8b05cdeda673bb78317
Author: David <Davidteri91@gmail.com>
Date:   Wed Aug 8 01:10:22 2018 +0200

    interfaces: fix error in build for android pie
    
    vendor/qcom/opensource/interfaces/wifi/supplicant/1.0/Android.mk: error: vendor.qti.hardware.wifi.supplicant-V1.0-java-static (JAVA_LIBRARIES android-arm64) missing android.hardware.wifi.supplicant-V1.0-java-static (JAVA_LIBRARIES android-arm64)
    You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
    vendor/qcom/opensource/interfaces/wifi/supplicant/1.0/Android.mk: error: vendor.qti.hardware.wifi.supplicant-V1.0-java-static (JAVA_LIBRARIES android-arm64) missing android.hidl.base-V1.0-java-static (JAVA_LIBRARIES android-arm64)
    You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
    vendor/qcom/opensource/interfaces/wifi/supplicant/1.1/Android.mk: error: vendor.qti.hardware.wifi.supplicant-V1.1-java-static (JAVA_LIBRARIES android-arm64) missing android.hardware.wifi.supplicant-V1.0-java-static (JAVA_LIBRARIES android-arm64)
    You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
    vendor/qcom/opensource/interfaces/wifi/supplicant/1.1/Android.mk: error: vendor.qti.hardware.wifi.supplicant-V1.1-java-static (JAVA_LIBRARIES android-arm64) missing android.hidl.base-V1.0-java-static (JAVA_LIBRARIES android-arm64)
    You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
    
    Signed-off-by: David Viteri <davidteri91@gmail.com>

diff --git a/wifi/supplicant/1.0/Android.mk b/wifi/supplicant/1.0/Android.mk
index 5b0144d..acfed2c 100644
--- a/wifi/supplicant/1.0/Android.mk
+++ b/wifi/supplicant/1.0/Android.mk
@@ -91,10 +91,15 @@ intermediates := $(call local-generated-sources-dir, COMMON)
 
 HIDL := $(HOST_OUT_EXECUTABLES)/hidl-gen$(HOST_EXECUTABLE_SUFFIX)
 
+ifeq (1,$(filter 1,$(shell echo "$$(( $(PLATFORM_SDK_VERSION) >= 28 ))" )))
+LOCAL_STATIC_JAVA_LIBRARIES := \
+    android.hardware.wifi.supplicant-V1.0-java \
+    android.hidl.base-V1.0-java
+else
 LOCAL_STATIC_JAVA_LIBRARIES := \
     android.hardware.wifi.supplicant-V1.0-java-static \
-    android.hidl.base-V1.0-java-static \
-
+    android.hidl.base-V1.0-java-static
+endif
 
 #
 # Build ISupplicantVendorStaIface.hal
